python/SetupPython.cmake

```diff
@@ -9,7 +9,7 @@ set(Python_USE_STATIC_LIBS OFF)
 if (PYTHON_VERSION)
   find_package(Python ${PYTHON_VERSION} EXACT REQUIRED COMPONENTS Interpreter Development OPTIONAL_COMPONENTS NumPy)
 else()
-  find_package(Python REQUIRED COMPONENTS Interpreter Development OPTIONAL_COMPONENTS NumPy)
+  find_package(Python 3 EXACT REQUIRED COMPONENTS Interpreter Development OPTIONAL_COMPONENTS NumPy)
```

Comment:
Force finding python3, not python2, related to SWIG using py3 now. Python 2 is EOL since 2020 anyways, we do not want to support it.